bundles/binding/org.openhab.binding.akm868/src/main/java/org/openhab/binding/akm868/internal/AKM868GenericBindingProvider.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.akm868.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.openhab.binding.akm868.AKM868BindingProvider;
+import org.openhab.core.binding.BindingConfig;
+import org.openhab.core.items.Item;
+import org.openhab.model.item.binding.AbstractGenericBindingProvider;
+import org.openhab.model.item.binding.BindingConfigParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * This class is responsible for parsing the binding configuration.
+ * 
+ * @author Michael Heckmann
+ * @since 1.7.0
+ */
+public class AKM868GenericBindingProvider extends AbstractGenericBindingProvider implements AKM868BindingProvider {
+
+	static final Logger logger = LoggerFactory.getLogger(AKM868GenericBindingProvider.class);
+	private Map<String, Item> items = new HashMap<String, Item>();
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	public String getBindingType() {
+		return "akm868";
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	public void validateItemType(Item item, String bindingConfig) throws BindingConfigParseException {
+		//if (!(item instanceof SwitchItem || item instanceof DimmerItem)) {
+		//	throw new BindingConfigParseException("item '" + item.getName()
+		//			+ "' is of type '" + item.getClass().getSimpleName()
+		//			+ "', only Switch- and DimmerItems are allowed - please check your *.items configuration");
+		//}
+//		logger.debug("AKM868GenericBindingProvider: validateItem");
```

Comment:
i would expect your binding to only accept Switch items right? Then you should active this validation again